docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2305 @@
+    a strategy for splitting those before you generate your embeddings.
+
+
+Create the schema to store embeddings
+=====================================
+
+To be able to store data in the database, we have to create its schema first.
+We want to make the schema as simple as possible and store only the relevant
+data. We need to store the section's embeddings, content, and the number of
+tokens. The embeddings allow us to match content to questions. The content
+gives us context to feed to the LLM. We will need the token count later when
+calculating how many related sections fit inside the prompt context while
+staying under the model's token limit.
+
+Open the empty schema file that was generated when we initialized the EdgeDB
+project (located at ``dbschema/default.esdl`` from the project directory).
+We'll walk through what we'll add to it, one step at a time. First, add this at
+the top of the file (above ``module default {``):
+
+.. code-block:: sdl
+    :caption: dbschema/default.esdl
+
+    using extension pgvector;
+
+We are able to store embeddings and find similar embeddings in the EdgeDB
+database because of the ``pgvector`` extension. In order to use it in our
+schema, we have to activate the ``ext::pgvector`` module with ``using extension
+pgvector`` at the beginning of the schema file. This module gives us access to
+the ``ext::pgvector::vector`` data type as well as few similarity functions and
+indexes we can use later to retrieve embeddings. Read our `pgvector
+documentation <https://www.edgedb.com/docs/stdlib/pgvector>`_ for more details
+on the extension.
+
+Just below that, we can start building our module by creating a new scalar
+type.
+
+.. code-block:: sdl
+    :caption: dbschema/default.esdl
+
+    …
+    module default {
+      scalar type OpenAIEmbedding extending
+        ext::pgvector::vector<1536>;
+
+      type Section {
+        # We will build this out next
+      }
+    }
+
+With the extension active, we may now add vector properties when defining our
+type. However, in order to be able to use indexes, the vectors in question need
+to be a of a fixed length. This can be achieved by creating a custom scalar
+extending the vector and specifying the desired length. OpenAI embeddings have
+length of 1,536, so that's what we use in our schema for this custom scalar.
+
+Now, the ``Section`` type:
+
+.. code-block:: sdl
+    :caption: dbschema/default.esdl
+
+    …
+      type Section {
+        required content: str;
+        required tokens: int16;
+        required embedding: OpenAIEmbedding;
+
+        index ext::pgvector::ivfflat_cosine(lists := 3)
+          on (.embedding);
+      }
+    …
+
+The ``Section`` contains properties to store the content, a count of tokens,
+and the embedding, which is of the custom scalar type we created in the
+previous step.
+
+We've also added an index inside the ``Section`` type to speed up queries. In
```

Comment:
I think we should remove index completely, our guide has 3 files :D 
But we should mention that it should be used when there are lots of files, I don't really know when the results start being visible? @jaclarke 
@vpetrovykh is on vacation, maybe @elprans can help? Do you know when it makes sense to add index, after 100 files, 1000, 10000?

Reply:
There's nothing wrong with adding an index on a small table. I would leave this as-is.